backend/test/v2/api/pipeline_upload_api_test.go

```diff
@@ -0,0 +1,282 @@
+var createdPipelines []*model.V2beta1Pipeline
+var testStartTime strfmt.DateTime
+var uploadParams *upload_params.UploadPipelineParams
+var pipelineFilesRootDir = utils.GetPipelineFilesDir()
+
+// ####################################################################################################################################################################
+// ################################################################### SET AND TEARDOWN ################################################################################
+// ####################################################################################################################################################################
+
+var _ = BeforeEach(func() {
+	logger.Log("################### Setup before each test #####################")
+	testStartTime, _ = strfmt.ParseDateTime(time.Now().Format(time.DateTime))
+	createdPipelines = []*model.V2beta1Pipeline{}
+	expectedPipeline = new(model.V2beta1Pipeline)
+	expectedPipeline.CreatedAt = testStartTime
+	pipelineGeneratedName = "apitest-" + strconv.FormatInt(time.Now().UnixNano(), 10)
+	uploadParams = upload_params.NewUploadPipelineParams()
+	expectedPipeline.Namespace = *namespace
+})
+
+var _ = AfterEach(func() {
+	// Delete pipelines created during the test
+	logger.Log("################### Cleanup after each test #####################")
+	logger.Log("Deleting %d pipeline(s)", len(createdPipelines))
+	for _, pipeline := range createdPipelines {
+		utils.DeletePipeline(pipelineClient, pipeline.PipelineID)
+	}
+})
+
+// ####################################################################################################################################################################
+// ################################################################### TESTS ################################################################################
+// ####################################################################################################################################################################
+
+// ################################################################################################################
+// ########################################################## POSITIVE TESTS ######################################
+// ################################################################################################################
+
+var _ = Describe("Verify Pipeline Upload >", Label("Positive", "PipelineUpload", "ApiServerTests", FullRegression), func() {
+
+	/* Critical Positive Scenarios of uploading a pipeline file */
+	Context("Upload a valid critical pipeline file and verify pipeline metadata after upload >", Label(Smoke), func() {
+		var pipelineDir = "valid/critical"
+		criticalPipelineFiles := utils.GetListOfFileInADir(filepath.Join(pipelineFilesRootDir, pipelineDir))
+		for _, pipelineFile := range criticalPipelineFiles {
+			It(fmt.Sprintf("Upload %s pipeline", pipelineFile), func() {
```

Comment:
See https://onsi.github.io/ginkgo/#dynamically-generating-specs
```suggestion
			pipelineFile := pipelineFile
			It(fmt.Sprintf("Upload %s pipeline", pipelineFile), func() {
```

Reply:
Done